[Tooltips] Fixed positioning of tooltips.

Odyseus opened the pull request:
Currently, tooltips created with the **new Tooltips.Tooltip()** constructor are aligned to the top-left corner of the mouse pointer. This leads to having tooltips overlapped by the mouse pointer. And the bigger the mouse pointer, more of the tooltip is covered by it.

This commit aligns the tooltip to the bottom-right corner of the mouse pointer (approximately), reducing the possibility of the mouse pointer to overlap the tooltip.

#### Before
![Tooltips broken](https://cloud.githubusercontent.com/assets/3822556/19834781/cc097aa0-9e4c-11e6-92f7-19be6e221edb.png)

#### After
![Tooltips fixed](https://cloud.githubusercontent.com/assets/3822556/19880320/e85ce7c0-9fda-11e6-98e0-3e954bcb7c55.png)
### Some technicalities
- This change only affects tooltips created with the **new Tooltips.Tooltip()** constructor.
- As I didn't want to use a fixed size, I used the global preference _cursor-size_.

collinss commented:
This looks out of place to me, and could potentially be confusing to the average user in some situations. It might work better if you place it above or to the right instead. That's just a suggestion, though. I'm not exactly a UI expert.

Odyseus commented:
Hello, @collins.

I'm not an UI expert either, but I have chosen the current positioning by observing how the tooltips of several applications work. And what I observed was that all applications have completely different approaches.

The tooltips of the Gtk2 applications that I checked (Gimp and InkScape) look like this...

![ttgimp1](https://cloud.githubusercontent.com/assets/3822556/19878789/b0352dbc-9fcb-11e6-926a-239e45496d44.png) ![ttgimp2](https://cloud.githubusercontent.com/assets/3822556/19878790/b06e3846-9fcb-11e6-8986-78be5ffc4101.png)

...and they don't consider the mouse cursor size.

Qt applications are a completely messed up universe (LOL). Most of the ones that I tried have custom tooltips. Others have the same tooltips as the ones used by Gtk2 applications.

And finally, Gtk3 applications. Almost all Gtk3 applications position their tooltips related to the UI elements that they belong and center aligned. Like this on Nemo's UI...

![ttnemo2](https://cloud.githubusercontent.com/assets/3822556/19878791/b07624c0-9fcb-11e6-88ae-175b8ccba964.png)

But inside Nemo views, the tooltips mostly look like this...

![ttnemo1](https://cloud.githubusercontent.com/assets/3822556/19878792/b077c730-9fcb-11e6-8fbe-23936305d69b.png)

Seeing that all application handle tooltips completely different:

- Gtk2 tooltips are too close to the cursor and a big mouse cursor overlaps the tooltip.
- Qt tooltips are a mess or the obey Gtk2 rules.
- Gtk3 tooltips are too complex to implement (aligned to the element).
- Tooltips on Nemo's internal views are to far away from the mouse cursor.

I opted to use the one implementation that I liked the most and made more sense to me. Which is the tooltips used by Firefox...

![ttfirefox1](https://cloud.githubusercontent.com/assets/3822556/19878788/b0312050-9fcb-11e6-8280-2a970240f370.png)

Although, seeing that the *trend* is for the tooltip to be aligned diagonally with respect to the cursor, I could make them look like Nemo tooltips, like this...

![tttweaked](https://cloud.githubusercontent.com/assets/3822556/19880320/e85ce7c0-9fda-11e6-98e0-3e954bcb7c55.png)

Placing the tooltip above the cursor wouldn't make much sense. All applications across all toolkits display their tooltips bellow the cursor, unless there is no room, then they are displayed above the cursor. And placing them to the right of the cursor would present the original problem. When there is no room to the right to fit the tooltip, the tooltip is moved to the left and the cursor overlaps the tooltip.

I know that this entire commit is more of a workaround than a solution. Mainly because the *cursor-size* setting isn't the actual size of the cursor. But it's an easy fix that would improve the readability of Cinnamon's tooltips in most use cases.

Odyseus commented:
I updated this pull request.
- I changed the code so the tooltips are now aligned diagonally with respect to the mouse pointer.
- I added a more detailed commit message.
- I edited my first comment to reflect all of this changes.
